tabs/tab_data.py

```
@@ -228,12 +221,13 @@ def _():
                     unique_vals = new_df[col].dropna().unique()
                     is_numeric = pd.api.types.is_numeric_dtype(new_df[col])
                     if is_numeric and len(unique_vals) > 10:
-                         current_meta[col] = {'type': 'Continuous', 'map': {}, 'label': col}
+                        current_meta[col] = {'type': 'Continuous', 'map': {}, 'label': col}
                     else:
-                         current_meta[col] = {'type': 'Categorical', 'map': {}, 'label': col}
+                        current_meta[col] = {'type': 'Categorical', 'map': {}, 'label': col}
             
             var_meta.set(current_meta)
             ui.notification_show(f"✅ Loaded {len(new_df)} rows", type="message")
+            logger.info(f"✅ File uploaded: {f['name']} with {len(new_df)} rows")
```

Ruff (0.14.10)

Logging statement uses f-string

(G004)